scio-core/src/main/scala/com/spotify/scio/values/BloomFilter.scala

```diff
@@ -0,0 +1,202 @@
+   * for an element that was not actually present the colleciton from which this [[BloomFilter]]
+   * was built.
+   *
+   * Ideally, this number should be close to the `fpProb` parameter passed to the [[BloomFilter#apply]]
+   * when this filter was built or smaller. If it is significantly higher, it is usually the
+   * case that too many elements (more than expected) were present in the original collection
+   * from which this [[BloomFilter]] was built.
+   */
+  val expectedFpp: Double = internal.expectedFpp()
+
+  /**
+   * Returns an estimate for the total number of distinct elements that have been added to this
+   * [[BloomFilter]]. This approximation is reasonably accurate if it does not exceed the value of
+   * `expectedInsertions` that was used when constructing the filter.
+   */
+  val approximateElementCount: Long = internal.approximateElementCount()
+
+  /**
+   * Check approximate presence of an element in the BloomFilter
+   *
+   * @return true if the element may be present, false if it is definitely not present.
+   */
+  def mayBeContains(t: T): Boolean = internal.mightContain(t)
+
+  // Java Serialization
+  private def writeObject(out: ObjectOutputStream): Unit = {
+    out.writeObject(funnel)
+    internal.writeTo(out)
+  }
+
+  private def readObject(in: ObjectInputStream): Unit = {
+    val funnel = in.readObject().asInstanceOf[Funnel[T]]
+    val internal = gBloomFilter.readFrom(in, funnel)
+    BloomFilter.setField("funnel", funnel)
+    BloomFilter.setField("internal", internal)
+  }
+
+  /**
+   * Add an element to this [[BloomFilter]]. It creates a copy of the underlying
+   * structure.
+   *
+   * For creating BloomFilters from large collections, use [[BloomFilter#apply(Iterable)]]
+   * instead.
+   */
+  @experimental
+  def put(t: T): BloomFilter[T] = {
+    val copyOfInternal = internal.copy()
+    copyOfInternal.put(t)
+    this.copy(internal = copyOfInternal)
+  }
+
+  /**
+   * Add an Itearable of elements to this [[BloomFilter]]. It creates a copy of the underlying
+   * structure.
+   *
+   * For creating BloomFilters from large collections, use [[BloomFilter#apply(Iterable)]]
+   * instead.
+   */
+  @experimental
+  def putAll(elements: Iterable[T]): BloomFilter[T] = {
+    val copyOfInternal = internal.copy()
+    val it = elements.iterator
+    while (it.hasNext) {
+      copyOfInternal.put(it.next())
+    }
+    this.copy(internal = copyOfInternal)
+  }
+}
+
+object BloomFilter extends ApproxFilterCompanion {
+
+  /**
+   * Constructor for [[BloomFilter]]
+   *
+   * @param iterable An iterable of elements to be stored in the BloomFilter
+   * @param fpProb allowed false positive probability
+   * @param f [[Funnel]] for the type [[T]]
+   * @return A [[BloomFilter]] for the given [[Iterable]] of elements.
+   */
+  def apply[T](
+    iterable: Iterable[T],
+    fpProb: Double
+  )(
+    implicit f: Funnel[T]
+  ): BloomFilter[T] = {
+    val numElements = iterable.size
```

Comment:
When used with Scio, `Iterable[T]` can very likely be the lazy ones after `groupBy` so `.size` is O(n).
IMHO the default choice or one most likely picked by user (e.g. least # of args) should minimize risk of misuse.

OTOH `fpp` of 0.01 should be a sane default, since it has the best trade off between size and fpp?